### PR TITLE
ffi: cleanup structseq

### PR DESCRIPTION
**Pull request by deantvv**
For #1289

### CODE REVIEW

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:36]:
```suggestion
    #[cfg(not(PyPy))]
    #[cfg_attr(docsrs, doc(cfg(not(PyPy))))]
    pub fn PyStructSequence_NewType(desc: *mut PyStructSequence_Desc) -> *mut PyTypeObject;
```

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:58]:
```suggestion
    #[cfg(not(PyPy))]
    #[cfg_attr(docsrs, doc(cfg(not(PyPy))))]
    pub fn PyStructSequence_SetItem(arg1: *mut PyObject, arg2: Py_ssize_t, arg3: *mut PyObject);
    #[cfg(not(PyPy))]
    #[cfg_attr(docsrs, doc(cfg(not(PyPy))))]
    pub fn PyStructSequence_GetItem(arg1: *mut PyObject, arg2: Py_ssize_t) -> *mut PyObject;
```

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:3]:
```suggestion
#[cfg(not(Py_LIMITED_API))]
use crate::ffi::tupleobject::{PyTupleObject, PyTuple_GET_ITEM, PyTuple_SET_ITEM};
```

**Review by davidhewitt — COMMENTED** [review]
Thanks, looks great! Just a few tweaks needed before merging...

File: src/ffi/structseq.rs
Comment (deantvv) [reply src/ffi/structseq.rs:39]: Thanks for always catching these!

File: src/ffi/structseq.rs
Comment (deantvv) [reply src/ffi/structseq.rs]: Thanks!

File: src/ffi/structseq.rs
Comment (davidhewitt) [reply src/ffi/structseq.rs:4]: Hmm, this import is still failing on PyPy. It's probably tidier to remove this and use the definitions inline below?

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:45]:
```suggestion
pub type PyStructSequence = crate::ffi::PyTupleObject;
```

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:50]:
```suggestion
    crate::ffi::PyTuple_SET_ITEM(op, i, v)
```

File: src/ffi/structseq.rs
Comment (davidhewitt) [inline src/ffi/structseq.rs:56]:
```suggestion
    crate::ffi::PyTuple_GET_ITEM(op, i)
```

**Review by davidhewitt — COMMENTED** [review]
Hmmm looks like we've still got a bit of awkwardness on PyPy...

**Comment by deantvv** [comment]
> 
> 
> Hmmm looks like we've still got a bit of awkwardness on PyPy...

Resolved. Hope this time we can get a green light in CI.

**Review by davidhewitt — APPROVED** [review]
Hooray looks like we're finally there! Thanks again! 🚀

**Pull request merged**
